src/content.js

```diff
@@ -0,0 +1,168 @@
+import fs from "node:fs";
+import path from "node:path";
+import github from "@actions/github";
+import flatten from "flat";
+import yaml from "js-yaml";
+import markup from "markup-js";
+import Config from "./config.js";
+import SlackError from "./errors.js";
+
+/**
+ * The parsed payload provided to the action and passed to the preferred method
+ * of sending a payload.
+ */
+export default class Content {
+  /**
+   * Gather content from the provided payload or payload file path with parsings.
+   * @param {Config} config
+   * @returns {this} - An instance of this Content class.
+   */
+  get(config) {
+    switch (true) {
+      case !!config.inputs.payload && !!config.inputs.payloadFilePath:
+        throw new SlackError(
+          config.core,
+          "Invalid input! Just the payload or payload file path is required.",
+        );
+      case !!config.inputs.payload:
+        this.values = this.getContentPayload(config);
+        break;
+      case !!config.inputs.payloadFilePath:
+        this.values = this.getContentPayloadFilePath(config);
+        break;
+      default:
+        config.core.debug(
+          "Missing payload so gathering inputs from action context.",
+        );
+        this.values = github.context;
+        break;
+    }
+    if (config.inputs.payloadDelimiter) {
+      this.values = flatten(this.values, {
+        delimiter: config.inputs.payloadDelimiter,
+      });
+      for (const key of Object.keys(this.values)) {
+        this.values[key] = `${this.values[key]}`;
+      }
+    }
+    return this;
+  }
+
+  /**
+   * Format request content from payload values for use in the request.
+   * @param {Config} config
+   * @throws if the input payload or payload file path is invalid JSON.
+   * @returns {Content} - the parsed JSON payload to use in requests.
+   */
+  getContentPayload(config) {
+    if (!config.inputs.payload) {
+      throw new SlackError(
+        config.core,
+        "Invalid input! No payload content found",
+      );
+    }
+    try {
+      const input = this.templatize(config, config.inputs.payload);
+      const content = /** @type {Content} */ (
+        yaml.load(input, {
+          schema: yaml.JSON_SCHEMA,
+        })
+      );
+      return /** @type {Content} */ (content);
+    } catch (error) {
+      if (error instanceof Error) {
+        config.core.debug("Failed to parse input payload as YAML");
+        config.core.debug(error.message);
+      }
+    }
+    try {
+      const trimmed = config.inputs.payload.trim();
+      if (
+        !config.inputs.payload.startsWith("{") &&
+        !config.inputs.payload.endsWith("}")
+      ) {
+        config.core.debug(
+          "Wrapping input payload in braces to create valid JSON",
+        );
+        const comma = trimmed.replace(/,$/, ""); // remove trailing comma
+        const wrap = `{${comma}}`;
+        return JSON.parse(wrap);
+      }
+      return JSON.parse(trimmed);
+    } catch (error) {
```

Comment:
What's this `catch` look like for e.g. an invalid JSON string? Potentially we could be reporting errors in three ways in here: via `core.error`, via `console.error` and then `throw`ing the error. And isn't a side effect of constructing a `SlackError` that the error also gets passed once more to `core.error`?

Reply:
Exploring this brought solid improvements overall, thanks for calling it out! 🙌

Now, all errors are thrown as `SlackError`. And the `core.error` is used just from `index.js` on caught errors **or** logged errors with `@slack/web-api`. No more do we have `console.error`!

Here's an example workflow with some errors:

- https://github.com/zimeg/slack-github-action/actions/runs/11800944197/job/32873064898?pr=1#step:14:14
- https://github.com/zimeg/slack-github-action/actions/runs/11800944197/job/32873064898?pr=1#step:15:25

Also, the error catching is done **with** the [`cause`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error/cause) attribute where rethrowing is done:

https://github.com/zimeg/slack-github-action/blob/4f678e0219be124734ed6eaad400268b36f39d28/src/content.js#L92-L100

Although I'm unsure how this might be also included in the logs above... 🔍